.github/workflows/verify_span_metadata_markdown_is_updated.yml

```diff
@@ -0,0 +1,34 @@
+name: Verify docs/span_metadata.md is updated
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  verify_source_generators:
+    runs-on: windows-latest
+
+    steps:
+      - name: Support longpaths
+        run: git config --system core.longpaths true
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.100'
+
+      - name: "Regenerate docs/span_metadata.md"
+        run: .\tracer\build.ps1 GenerateSpanDocumentation
+
+      - name: "Verify no changes in docs/span_metadata.md"
+        run: |
+          git diff --quiet -- .\docs\span_metadata.md
+          if ($LASTEXITCODE -eq 1) {
+            git diff -- .\docs\span_metadata.md
+            Write-Error "Found changes in docs/span_metadata.md file. Run build task GenerateSpanDocumentation to regenerate the file with the latest C# rules."
+            Exit 1
```

Comment:
Do you plan to make this step mandatory in the repo?

Reply:
Hmm I can update our Automatic Instrumentation docs to add this step for the integration development, as that's the only time you'll usually need to update this. Implemented in c59488f6c51f1a71981aadcb4190b1dd81c82fde

Reply:
I suspect Pierre meant marking the GitHub action as required for merges 🙂

Reply:
> I suspect Pierre meant marking the GitHub action as required for merges 🙂

Yep. Sorry if it wasn't clear